api/client/stack/common.go

```diff
@@ -48,3 +50,43 @@ func getNetworks(
 		ctx,
 		types.NetworkListOptions{Filters: getStackFilter(namespace)})
 }
+
+type stack struct {
+	// Name is the name of the stack
+	Name string
+}
+
+func getStacks(
+	ctx context.Context,
+	apiclient client.APIClient,
+) ([]*stack, error) {
+
+	filter := filters.NewArgs()
+	filter.Add("label", labelNamespace)
+
+	services, err := apiclient.ServiceList(
+		ctx,
+		types.ServiceListOptions{Filter: filter})
+	if err != nil {
+		return nil, err
+	}
+	var stacks []*stack
+	m := make(map[string]struct{}, 0)
+	for _, service := range services {
+		labels := service.Spec.Labels
+		name, ok := labels[labelNamespace]
+		if !ok {
+			return nil, fmt.Errorf("cannot get label %s for service %s",
+				labelNamespace, service.ID)
+		}
+		_, ok = m[name]
+		if !ok {
+			stack := &stack{
+				Name: name,
+			}
```

Comment:
I wonder if it would be valuable to include a count of services in the stack as another field.